account_bank_statement_import_guess_partner/models/account_bank_statement_import.py

```diff
@@ -27,26 +27,30 @@ def _complete_transaction(self, transaction):
         sale_order_model = self.env["sale.order"]
         invoice = None
         transaction_keys = ["ref", "name"]
-        invoice_fields = ["invoice_origin", "ref", "name"]
+        invoice_fields = ["transaction_id", "invoice_origin", "ref", "name"]
         sale_order_fields = ["client_order_ref", "name"]
         for key in transaction_keys:
             value = transaction.get(key, False)
             if not value:
                 continue
             for fieldname in invoice_fields:
+                if not hasattr(invoice_model, fieldname):
+                    continue
                 invoice = invoice_model.search([(fieldname, "=", value)], limit=1)
                 if invoice:
                     # We need a partner of type contact here because of reconciliation.
                     # Lines cannot be reconciled if partner is an address
                     partner = self._get_effective_partner(invoice.partner_id)
                     transaction["partner_id"] = partner.id
-                    return
+                    break
```

Comment:
Why the break? The idea is that if we already have found the partner through the invoice, we no longer need to search on the Sale Order.

Reply:
You are right, this ended up there while I was testing for a solution. And it truly looks that the solution is the addition of this new key. Will retest and amend this one.